aten/src/ATen/core/op_registration/op_registration.h

```diff
@@ -317,7 +319,10 @@ class CAFFE2_API RegisterOperators final {
       return std::move(*this).kernel(
         std::move(dispatch_key),
         nullptr,
-        nullptr,  // setting cache creator to nullptr so calling the kernel doesn't need to call it, which would be expensive
+        // setting cache creator to nullptr so calling the kernel doesn't need to call it, which would be expensive
+        // This, however, only works if there are no constructor parameters (i.e. no runtime function pointer)
+        // Backend extensions use runtime function pointers, so we need a cache if sizeof...(ConstructorParameters) != 0
+        (sizeof...(ConstructorParameters) == 0) ? KernelCacheCreatorFunction(nullptr) : detail::KernelFactory<KernelFunctor, guts::decay_t<ConstructorParameters>...>(std::forward<ConstructorParameters>(constructorParameters)...),
```

Comment:
So, this patch also adds support for some operators which didn't work before?

Reply:
This is not for new operators but for backend extensions. All ATen operators in native_functions.yaml codegen registrations that don't need this, but if you look at "Additional concern: Extension backends" in https://github.com/pytorch/pytorch/issues/24132, you see the plan to keep globalATenDispatch().registerOp() working until we moved all ops to c10, and only then break all ops at once for them. To keep globalATenDispatch().registerOp() working, we need this.